app/email-preferences/page.tsx

```
@@ -0,0 +1,455 @@
+
+  // State for preferences (email type ID → selected frequency)
+  const [preferences, setPreferences] = useState<Record<string, EmailFrequency>>({});
+
+  // Initialize preferences based on EMAIL_TYPES
+  useEffect(() => {
+    const initialPreferences: Record<string, EmailFrequency> = {};
+
+    EMAIL_TYPES.forEach(emailType => {
+      // Default to the first available frequency or 'none'
+      initialPreferences[emailType.id] = emailType.frequencies[0] || 'none';
+    });
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)